src/psyclone/domain/lfric/lfric_kern_metadata.py

```diff
@@ -0,0 +1,676 @@
+        routine silently returns.
+
+        :raises: ParseError: if meta-data breaks inter-grid rules
+        '''
+        # pylint: disable=too-many-branches
+        # Dictionary of meshes associated with arguments (for inter-grid
+        # kernels). Keys are the meshes, values are lists of function spaces
+        # of the corresponding field arguments.
+        mesh_dict = OrderedDict()
+        # Whether or not any field args are missing the mesh_arg specifier
+        missing_mesh = False
+        # If this is an inter-grid kernel then it must only have field
+        # arguments. Keep a record of any non-field arguments for the benefit
+        # of a verbose error message.
+        non_field_arg_types = set()
+        const = LFRicConstants()
+        for arg in self._arg_descriptors:
+            # Collect info so that we can check inter-grid kernels
+            if arg.argument_type in const.VALID_FIELD_NAMES:
+                if arg.mesh:
+                    # Argument has a mesh associated with it so this must
+                    # be an inter-grid kernel
+                    if arg.mesh in mesh_dict:
+                        mesh_dict[arg.mesh].append(arg.function_space)
+                    else:
+                        mesh_dict[arg.mesh] = [arg.function_space]
+                else:
+                    # Record the fact that we have a field without a
+                    # mesh specifier (in case this is an inter-grid kernel)
+                    missing_mesh = True
+            else:
+                # Inter-grid kernels are only permitted to have field args
+                # so collect a list of other types
+                non_field_arg_types.add(arg.argument_type)
+
+        mesh_list = mesh_dict.keys()
+        if not mesh_list:
+            # There are no meshes associated with any of the arguments so
+            # this is not an inter-grid kernel
+            return
+
+        if len(const.VALID_MESH_TYPES) != 2:
+            # Sanity check that nobody has messed with the number of
+            # grid types that we recognise. This is here because the
+            # implementation assumes that there are just two grids
+            # (coarse and fine).
+            raise InternalError(
+                f"The implementation of inter-grid support in the LFRic "
+                f"API assumes there are exactly two mesh types but "
+                f"LFRicConstants.VALID_MESH_TYPES contains "
+                f"{len(const.VALID_MESH_TYPES)}: {const.VALID_MESH_TYPES}")
+        if len(mesh_list) != len(const.VALID_MESH_TYPES):
+            raise ParseError(
+                f"Inter-grid kernels in the Dynamo 0.3 API must have at least "
+                f"one field argument on each of the mesh types "
+                f"({const.VALID_MESH_TYPES}). However, kernel {self.name} has "
+                f"arguments only on {[str(name) for name in mesh_list]}")
```

Comment:
I would suggest single quotations around `'mesh_list'`.

Reply:
That turns it into a string, and `mesh_list` is being used as a list here I believe

Reply:
I see, thanks!